PurchasesTests/Mocks/MockOperationDispatcher.swift

```diff
@@ -58,7 +59,7 @@ class MockOperationDispatcher: OperationDispatcher {
             return
         }
         if shouldInvokeDispatchOnHTTPSerialQueueBlock {
-            block()
+            serialQueue.async(execute: block)
```

Comment:
@vegaro you were totally right, this has no guarantee of serial execution, we need to add a queue for that 😄

Reply:
Yeah, so I realized that https://github.com/RevenueCat/purchases-ios/pull/674/files?file-filters%5B%5D=.h&file-filters%5B%5D=.m&file-filters%5B%5D=.swift#diff-e0adaebb0b6acbb722fb5b795187c9f213ce6671a6e142e39a49341630680872R147 executes in a thread and it's not guaranteed that the blocks execute serially if doing just `block()` here, since they might execute in a thread. 

Alternatively to this approach, we could execute the `block()` in the main thread, but honestly, using another serial queue like this one mimics the real behavior of `HTTPClient` a bit more. Maybe we shouldn't use the `MockOperationDispatcher` and we should use a real `OperationDispatcher`? I am worried we miss some bugs if we use the mocked version...

Reply:
I think I've used the real operation dispatcher in a few tests because of this same thing. we can have a fancier mock, but the original class is so simple that a fancier mock would probably get more complex / bug-prone than the real thing

Reply:
I totally agree, an even better solution is to use the real operation dispatcher if the mock isn't useful 😄